core/src/main/java/com/linecorp/armeria/server/Http2ServerConnectionHandler.java

```diff
@@ -118,11 +124,54 @@ private void maybeInitializeKeepAliveHandler(ChannelHandlerContext ctx) {
         }
     }
 
-    private void destroyKeepAliveHandler() {
+    private void cancelScheduledTasks() {
+        gracefulConnectionShutdownHandler.cancel();
         keepAliveHandler.destroy();
     }
 
     KeepAliveHandler keepAliveHandler() {
         return keepAliveHandler;
     }
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        if (evt instanceof InitiateConnectionShutdown) {
+            setGoAwayDebugMessage("app-requested");
```

Comment:
Should this be `client-requested`?

Reply:
It's requested by the server application, not by client. I picked this debug msg following example in grpc-netty https://github.com/grpc/grpc-java/blob/a282019902031faf27e9ff2de1e3ae6ddb1f4954/netty/src/main/java/io/grpc/netty/NettyServerHandler.java#L651

Reply:
Ah, I see. Then it's fine.